entityx/help/Pool_test.cc

```diff
@@ -13,8 +13,9 @@
 #include <vector>
 #include "entityx/3rdparty/catch.hpp"
 #include "entityx/help/Pool.h"
+#include "entityx/Entity.h"
 
-struct Position {
+struct Position : public entityx::Component<Position> {
```

Comment:
Interesting question. If I don't extend `Component<>` this code wouldn't compile for me. It couldn't find a definition for `Entity::remove<Position>`. However, if I use a component that extends `Component<>` then it accepts it happily.

I'm a little unsure if why this is, to be honest.

If components are supposed to be POD types, why have a base `Component<>` at all?